File: Taller 3/Python/tv.py

class TV:
    _numTV = 0

    def __init__(self, marca, estado):
        self._marca = marca
        self._canal = 1
        self._precio = 500
        self._estado = estado
        self._volumen = 1
        TV._numTV+= 1


    def setMarca(self, marca):
        self._marca = marca
    

    def setCanal(self, canal):
        if ((canal < 121 and canal > 0) and self._estado == True):
            self._canal = canal
    

    def setPrecio(self, precio):
        self._precio = precio


    def setVolumen(self, volumen):
        if (volumen <= 7 and volumen >= 0 and self._estado == True):
            self._volumen = volumen
    
    def setControl(self, control):
        self._control = control
    

    def getMarca(self):
        return self._marca
    

    def getCanal(self):
        return self._canal
    

    def getPrecio(self):
        return self._precio
    

    def getVolumen(self):
        return self._volumen
    
    def getControl(self):
        return self._control
    

    #------------------------------
    @classmethod
    def setNumTV(cls, num):
        cls._numTV = num

    @classmethod
    def getNumTV(cls):
        return cls._numTV
    
    #------------------------------

    #------------------------------
    def turnOn(self):
        self._estado = True
    

    def turnOff(self):
        self._estado = False
    
    def getEstado(self):
        return self._estado
    
    #------------------------------

    #------------------------------
    #Canal
    def canalUp(self):
        if (self._canal < 120 and self._estado == True):
            self._canal += 1
        
    def canalDown(self):
        if (self._canal > 1 and self._estado == True):
            self._canal -= 1
    

    #Volumen
    def volumenUp(self):
        if (self._volumen < 7 and self._estado == True):
            self._volumen += 1
            

    def volumenDown(self):
        if (self._volumen > 0 and self._estado == True):
            self._volumen -= 1
    #------------------------------

File: OOP/Taller 3/Java/TV.java
package taller3.televisores;

public class TV {

    private Marca marca;
    private int canal = 1;
    private int precio = 500;
    private boolean estado;
    private int volumen = 1;
    private Control control;
    private static int numTV = 0;

    public TV(Marca marca, boolean estado) {
        this.marca = marca;
        this.estado = estado;
        TV.numTV++;
    }

    public void setMarca(Marca marca) {
        this.marca = marca;
    }

    public void setCanal(int canal) {
        if ((canal < 121 && canal > 0) && this.estado == true) {
            this.canal = canal;
        }
    }

    public void setPrecio(int precio) {
        this.precio = precio;
    }

    public void setVolumen(int volumen) {
        if ((volumen < 8 && volumen >= 0) && this.estado == true) {
            this.volumen = volumen;
        }
    }

    public void setControl(Control control) {
        this.control = control;
    }

    public Marca getMarca() {
        return this.marca;
    }

    public int getCanal() {
        return this.canal;
    }

    public int getPrecio() {
        return this.precio;
    }

    public int getVolumen() {
        return this.volumen;
    }

    public Control getControl() {
        return this.control;
    }

    // ------------------------------
    public static void setNumTV(int num) {
        TV.numTV = num;
    }

    public static int getNumTV() {
        return TV.numTV;
    }
    // ------------------------------

    // ------------------------------
    public void turnOn() {
        this.estado = true;
    }

    public void turnOff() {
        this.estado = false;
    }

    public boolean getEstado() {
        return this.estado;
    }
    // ------------------------------

    // ------------------------------
    // Canal
    public void canalUp() {
        if (this.canal < 120 && this.estado == true) {
            this.canal += 1;
        }
    }

    public void canalDown() {
        if (this.canal > 1 && this.estado == true) {
            this.canal -= 1;
        }

    }

    // Volumen
    public void volumenUp() {
        if (this.volumen < 7 && this.estado == true) {
            this.volumen += 1;
        }
    }

    public void volumenDown() {
        if (this.volumen > 0 && this.estado == true) {
            this.volumen -= 1;
        }
    }
    // ------------------------------

}